lib/src/client.dart

```diff
@@ -1157,4 +1157,77 @@ class TwitchClient {
     );
     return TwitchTokenRefresh.fromJson(data);
   }
+
+  /// Create a clip
+  ///
+  /// `broadcasterId` required
+  /// `hasDelay` required
+  ///
+  /// documentation : https://dev.twitch.tv/docs/api/reference/#create-clip
+  Future<TwitchCreateClipResponse> createClip({required String broadcasterId, required bool hasDelay}) async {
+    final data = await twitchHttpClient.postCall<Map<String, dynamic>>(
+      ['clips'],
+      <String, Object>{},
+      queryParameters: {'broadcaster_id': broadcasterId, 'has_delay': hasDelay},
+    );
+    return TwitchCreateClipResponse.fromJson(data);
+  }
+
+  /// Get a clip
+  ///
+  /// `broadcasterId` exclusive Required An ID that identifies the broadcaster
+  /// `gameId` exclusive Required An ID that identifies the game whose clips you want to get
+  /// `ids` exclusive Required An ID that identifies the clip to get
+  /// `startedAt` format RFC3339
+  /// `endedAt`   format RFC3339
+  /// `first` The maximum number of clips to return per page in the response.
+  /// `before` The cursor used to get the previous page of results
+  /// `after` The cursor used to get the next page of results
+  /// `isFeatured`
+  ///
+  /// documentation : https://dev.twitch.tv/docs/api/reference/#get-clips
+  Future<TwitchClipResponse> getClips({
+    String? broadcasterId,
+    String? gameId,
+    List<String>? ids,
+    String? startedAt,
+    String? endedAt,
+    int first = 20,
+    String? before,
+    String? after,
+    bool? isFeatured,
+  }) async {
+    assert(broadcasterId != null && gameId != null && ids != null, "One of exclusive parameter must have a value");
+    assert(first > 0 && first < 101);
+    if (ids != null) {
+      assert(ids.length < 101);
+    }
+    String exclusiveParameter = "";
+    String exclusiveParameterValue = "";
+
+    if (broadcasterId?.isNotEmpty ?? false) {
+      exclusiveParameter = 'broadcaster_id';
+      exclusiveParameterValue = broadcasterId!;
+    } else if (gameId?.isNotEmpty ?? false) {
+      exclusiveParameter = 'game_id';
+      exclusiveParameterValue = gameId!;
+    } else if (ids?.isNotEmpty ?? false) {
+      exclusiveParameter = 'id';
+      exclusiveParameterValue = ids!.join(',');
+    }
```

Comment:
Prefer using pattern matching:
```suggestion
  final exclusiveParam = switch ((broadcasterId, gameId, ids)) {
    (final broadcasterId?, null, null) when broadcasterId.isNotEmpty => MapEntry('broadcaster_id', broadcasterId),
    (null, final gameId?, null) when gameId.isNotEmpty => MapEntry('game_id', gameId),
    (null, null, final ids?) when ids.isNotEmpty => MapEntry(id, ids.join(',')),
    _ => null,
  };
```
Then you'll be able to do the following:
```dart
queryParameters: {
  if (mapEntry != null) mapEntry.key: mapEntry.value,
  // ...
},
```

Reply:
i trying your code but not work to, so i keep the if,else but i use mapEntry now

Reply:
https://github.com/TesteurManiak/twitch_api_flutter/pull/117#discussion_r1770943314

Reply:
@TesteurManiak Ok i'll try it this evening

Reply:
Ok i'll try the changes this evening

Reply:
i'm pretty new to flutter / dart that why i don't know the basic stuff